api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/AssetGroupServiceImpl.java

```diff
@@ -863,4 +864,34 @@ private String getPluginAliasQuery(String datasource) {
 		}
 		return "";
 	}
+
+	/**
+	 * Retrieves a list of indices associated with a datasource.
+	 * Sends HTTP requests to Elasticsearch to fetch indices information.
+	 *
+	 * @return A list of indices associated with the specified indices.
+	 */
+	@Override
+	public List<String> fetchIndices(String datasource) {
+		List<String> indexNames = new ArrayList<>();
+		try {
+			String urlToQuery = "/_cat/indices/" + datasource + "_*";
+			Response response = commonService.invokeAPI("GET", urlToQuery, null);
+			if (response == null || response.getStatusLine().getStatusCode() != 200) {
+				log.error("Unable to fetch aliases for {} with response {}", datasource, response);
+				return null;
+			}
+			String indexData = EntityUtils.toString(response.getEntity());
+			String[] indexLines = indexData.split("\\r?\\n");
```

Comment:
try to add `?format=json` at the end of your URL and OS will return indexes in JSON format and you will have no need to parse them as a string

Reply:
since we can use wild card for removing indices, I removed this code